webapps/ui/cockpit/src/modules/components/TablePaginated/TablePaginated.js

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect, useState } from "react";
+import classNames from "classnames";
+import { Pagination, Table, LoadingIndicator } from "components";
+
+import "./TablePaginated.scss";
+
+const ZERO_NOOP = () => 0;
+const NOOP = () => {};
+
+/**
+ * A paginated table component.
+ *
+ * @param rowsPerPage is the number of rows which are displayed per page
+ * @param header is a {JSX.Element} which is placed inside the <thead>...</thead> of the table
+ * @param generateRow is a function which is called for each row, passes the row object and returns a {JSX.Element}
+ * @param fetchPage is a function which returns the page, passes the page number and returns an array of rows (can be a promise); each row must have a unique id
+ * @param fetchRowsCount is a function which returns the total amount of rows wrapped in a promise
+ * @param noRows is {JSX.Element} which is rendered when the rows count is zero
+ * @param onRowSelection is a function which is called when a row is selected and passes the selected rows
+ * @returns the TablePaginated component as {JSX.Element}
+ */
+function TablePaginated({
+  rowsPerPage = 5,
+  header,
+  generateRow,
+  fetchPage,
+  fetchRowsCount = ZERO_NOOP,
+  noRows = null,
+  onRowSelection = NOOP
+}) {
+  const [allSelected, setAllSelected] = useState(false);
+  const [page, setPage] = useState(1);
+  const [rows, setRows] = useState([]);
+  const [rowsCount, setRowsCount] = useState(0);
+  const [rowSelection, setRowSelection] = useState({});
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    if (fetchRowsCount !== ZERO_NOOP) {
+      fetchRowsCount().then(rowsCount => setRowsCount(rowsCount.count));
+    } else {
+      setRowsCount(1);
+    }
+  }, [fetchRowsCount]);
```

Comment:
The if-else should not be necessary, if `ZERO_NOOP = () => {return {count: 1}}`.